raster-foundry/raster-foundry
# Support Intercom conversations

## Overview

This PR supports intercom communications by having conversations with customers:
- add migration, data model, dao methods, and tests for previous conversation IDs and for new conversation IDs
- add intercom endpoint calls for creating a conversation and for replying to conversations
- in places where we dispatch intercom notifications, hook up the above two to work together:
    - create task grid
    - upload processing
    - stac exports

### Checklist

- ~Swagger specification updated~
- ~Any content changes are properly templated using `BUILDCONFIG.APP_NAME`~
- ~Any new endpoints have scope validation and are included in the integration test csv~

### Demo

All system messages appear in one conversation
![Screen Shot 2021-04-20 at 12 30 19 PM](https://user-images.githubusercontent.com/16109558/115432340-42282d00-a1d4-11eb-8888-2e7dfcfb71ba.png)

## Testing Instructions

- Run migration
- Assemble api jar and batch jar, rebuild batch service if need be
- In a batch shell, make sure running the following creates a conversation in intercom message with this dev user. Sending subsequent messages should all show in this conversation
```
java -cp /opt/raster-foundry/jars/batch-assembly.jar com.rasterfoundry.batch.Main notify-intercom <dev user ID> <message>
```
- Connect local GW to your local RF servers, create a campaign and upload an image, process the upload. After upload processing, the message should be sent to the same conversation as above
- Do a STAC export of a campaign. When export is ready, a message should be sent to the same conversation to the dev user as the above

Closes https://github.com/raster-foundry/groundwork/issues/1035
